File: core/src/io/github/jerukan/game/gameunits/unitdata/VillageUnit.java
package io.github.jerukan.game.gameunits.unitdata;

import io.github.jerukan.game.gameunits.Unit;
import io.github.jerukan.util.Assets;

public class VillageUnit extends BaseUnit {

    public VillageUnit() {
        name = "village";
        baseHealth = 20;

        baseCost = 0;
        setTexture(Assets.getTexture(Assets.village));
    }

    @Override
    public void onCreation(Unit self) {

    }

    @Override
    public void onDeath(Unit self) {

    }

    @Override
    public void onTurnStart(Unit self) {

    }

    @Override
    public void onTurnEnd(Unit self) {

    }
}

File: core/src/io/github/jerukan/game/gameunits/unitdata/VillageUnit.kt
package io.github.jerukan.game.gameunits.unitdata

import io.github.jerukan.game.GameState
import io.github.jerukan.game.Player
import io.github.jerukan.game.gameunits.Unit
import io.github.jerukan.game.gameunits.unitdata.unitactions.UnitAction
import io.github.jerukan.util.Assets
import io.github.jerukan.util.Position

class VillageUnit : BaseUnit() {
    init {
        name = "village"
        baseHealth = 20

        baseCost = 0

        actions = arrayOf()

        baseUpkeep = 0

        type = BaseUnit.Type.BUILDING

        description = "This is your lifeline. Don\'t let it die."
        texture = Assets.getTexture(Assets.village)
    }

    override fun _canBuild(pos: Position, owner: Player): Boolean {
        return canBuild(owner)
    }

    override fun canBuild(owner: Player): Boolean {
        return !GameState.instance.unitManager.playerHasUnit(owner, id)
    }

    override fun onCreation(self: Unit) {

    }

    override fun onDeath(self: Unit) {

    }

    override fun onTurnStart(self: Unit) {

    }

    override fun onTurnEnd(self: Unit) {

    }
}
